File: bug-zoo/species/BZ-SHAPE-005-null-boundary-equivalence/typescript/fixed/class-validator/harness/src/UserDirectory.ts
// @ts-nocheck
import { IsString } from "class-validator";

export class LookupRequest {
  @IsString()
  name: string;
}

export function lookup(input: LookupRequest): string {
  if (input.name == null) {
    throw new TypeError("name must be non-null");
  }
  return "user:" + input.name.toUpperCase();
}

File: menagerie/bug-zoo/species/BZ-SHAPE-005-null-boundary-equivalence/csharp/fixed/data-annotations/harness/src/UserDirectory.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace BugZoo.CSharpNullBoundary.DataAnnotations;

public sealed class LookupRequest
{
    [Required]
    public string Name { get; init; } = "";
}

public static class UserDirectory
{
    public static string Lookup(string? name)
    {
        if (name is null)
        {
            throw new ArgumentNullException(nameof(name), "name must be non-null");
        }

        return "user:" + name.ToUpperInvariant();
    }
}
